### PR TITLE
[maccel]: Change --maccel option from a string option to a list of enums

### PR DESCRIPTION
**Pull request by etiotto**
This PR resolve issue https://github.com/onnx/onnx-mlir/issues/1259 by changing the type for the `--maccel` option.
Previously the `--maccel` option would accept a string as its argument and fail to diagnose illegal/unsupported accelerators. With this PR we accept only valid/supported arguments: 

```
 % onnx-mlir --help |grep -A2 maccel                                                           
  --maccel=<value>                                   - Specify an accelerator to generate code for
    =NNPA                                            -   IBM Telum processor
 ```

When an invalid argument is supplied it is diagnosed during option processing with an hard error:
```
% onnx-mlir --O3 --EmitLib  --maccel="" simplemm.onnx                                     
onnx-mlir: for the --maccel option: Cannot find option named ''!
```

Signed-off-by: Ettore Tiotto <etiotto@ca.ibm.com>

### CODE REVIEW

File: src/Accelerators/Accelerator.hpp [hunk 1]
```diff
@@ -24,8 +24,17 @@ namespace accel {
 
 class Accelerator {
 public:
-  Accelerator();
+  /// Kinds of accelerators.
+  enum class Kind {
+    NNPA, // IBM Telum coprocessor
```
Comment (caoimhinuibrian) [inline src/Accelerators/Accelerator.hpp:29]: This exposes the accelerator names outside of their subdirectory. My idea for this is to have CMake generate the enum values in an include and so on ...

Comment (etiotto) [reply src/Accelerators/Accelerator.hpp:29]: This is just an enumerator so I do not understand the reason we do not want to "expose it".  It does not force the accelerator class type to be exposed.

Comment (caoimhinuibrian) [reply src/Accelerators/Accelerator.hpp:29]: Yes, but in my opinion, it is simpler to contain everything in the specific accelerator subdirectory and have CMake generate the "glue" based on the values specified in the environment variable. I can do it if you want, but if you feel strongly that this way is better I guess I can let it slide.

File: src/Compiler/CompilerUtils.cpp [hunk 2]
```diff
@@ -126,6 +124,12 @@ static llvm::cl::opt<std::string> march("march",
     llvm::cl::value_desc("Target a specific architecture type"),
     llvm::cl::cat(OnnxMlirOptions), llvm::cl::ValueRequired);
 
+llvm::cl::list<accel::Accelerator::Kind> maccel("maccel",
+    llvm::cl::desc("Specify an accelerator to generate code for"),
+    llvm::cl::values(clEnumValN(
+        accel::Accelerator::Kind::NNPA, "NNPA", "IBM Telum processor")),
+    llvm::cl::cat(OnnxMlirOptions), llvm::cl::ValueRequired);
```
Comment (tungld) [inline src/Compiler/CompilerUtils.cpp:131]: Can we make this optional? For example, it appears only when we build with some accelerators, i.e. `ACCELERATORS_TO_BUILD` is specified. 

If it is impossible, we need to let users know the current driver is not built with accelerator when they specify `--maccel=NNPA`

Comment (caoimhinuibrian) [reply src/Compiler/CompilerUtils.cpp:131]: @tungld @etiotto that's what my approach would do

File: src/Accelerators/Accelerator.hpp (hunk 1 above)
Comment (etiotto) [reply src/Accelerators/Accelerator.hpp:29]: @caoimhinuibrian do you have the cmake code to generate the include file you mentioned, happy to add it to the patch.

**Review by caoimhinuibrian — CHANGES_REQUESTED** [review]
I'd like to have a meeting to talk about this

**Review by tungld — APPROVED** [review]
LGTM

**Pull request merged**